### ISSUE TITLE
`never_loop` false positive with break from iterator

### ISSUE THREAD

**Issue by bupknar** [rust-lang-nursery/rust-clippy#3232]
This infinite loop is reported by Clippy to be never looping:
```rust
fn main() {
    loop {
        for _ in std::iter::empty::<()>() {
            break;
        }
    }
}
```
Clippy says:
```
error: this loop never actually loops
 --> src/main.rs:3:9
  |
3 | /         for _ in std::iter::empty::<()>() {
4 | |             break;
5 | |         }
  | |_________^
  |
  = note: #[deny(never_loop)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#never_loop
```
Clippy version: 0.0.212

**Comment by bupknar** [comment]
Whoops, I'm a dummy. I just now realized Clippy is talking about the inner loop, which actually _does_ never loop. So Clippy is right! Thank you Clippy! :smile:
